### ISSUE TITLE
TracerWarning while Pytorch2ONNX conversion. 'Converting a tensor to a Python boolean might cause the trace to be incorrect.'. 

### ISSUE THREAD

**Issue by pingguoyouzidaxigua** [open-mmlab/mmsegmentation#903]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**
There is a warning when fcn_unet_s5-d16_128x128_40k_chase_db1 exports onnx, and the exported onnx sees a large number of duplicate structures in netron, which is not the problem with the --cfg-options-model.test_cfg.mode-"whole" mode

**Reproduction**

1. What command or script did you run?

python pytorch2onnx work_dirs/fcn_unet_s5-d16_128x128_40k_chase_db1/fcn_unet_s5-d16_128x128_40k_chase_db1.py
--checkpoint=work_dirs/fcn_unet_s5-d16_128x128_40k_chase_db1/epoch_2.pth
--shape=128

2. Did you make any modifications on the code or config? Did you understand what you have modified?
3. What dataset did you use?
chase_db1
**Environment**
`sys.platform: linux
Python: 3.8.11 (default, Aug  3 2021, 15:09:35) [GCC 7.5.0]
CUDA available: True
GPU 0: GeForce RTX 3080
CUDA_HOME: None
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.8.1
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.3-Product Build 20210617 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.7.0 (Git Hash 7aed236906b1f7a05c0917e5257a1af05e9ff683)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.8.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.9.1
OpenCV: 4.5.3
MMCV: 1.3.13
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 11.1
MMSegmentation: 0.17.0+cae715a`
1. Please run `python mmseg/utils/collect_env.py` to collect necessary environment infomation and paste it here.
2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**
`/home/lsh/mmsegmentation/mmseg/models/segmentors/encoder_decoder.py:166: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  h_grids = max(h_img - h_crop + h_stride - 1, 0) // h_stride + 1
/home/lsh/mmsegmentation/mmseg/models/segmentors/encoder_decoder.py:167: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  w_grids = max(w_img - w_crop + w_stride - 1, 0) // w_stride + 1
/home/lsh/mmsegmentation/mmseg/models/segmentors/encoder_decoder.py:174: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  y2 = min(y1 + h_crop, h_img)
/home/lsh/mmsegmentation/mmseg/models/segmentors/encoder_decoder.py:175: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  x2 = min(x1 + w_crop, w_img)
/home/lsh/mmsegmentation/mmseg/models/segmentors/encoder_decoder.py:181: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  (int(x1), int(preds.shape[3] - x2), int(y1),
/home/lsh/mmsegmentation/mmseg/models/segmentors/encoder_decoder.py:182: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  int(preds.shape[2] - y2)))
/home/lsh/mmsegmentation/mmseg/models/segmentors/encoder_decoder.py:185: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert (count_mat == 0).sum() == 0
/home/lsh/mmsegmentation/mmseg/models/segmentors/encoder_decoder.py:189: TracerWarning: Converting a tensor to a NumPy array might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  count_mat.cpu().detach().numpy()).to(device=img.device)
/home/lsh/mmsegmentation/mmseg/models/segmentors/encoder_decoder.py:188: TracerWarning: torch.from_numpy results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  count_mat = torch.from_numpy(
Successfully exported ONNX model: tmp.onnx`

Process finished with exit code 0
```



**Bug fix**

If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by VVsssssk** [comment]
Hello,  if you don't set  --cfg-options-model.test_cfg.mode-"whole", you will use slide inference model,  inference by sliding window with overlap, referring to https://github.com/open-mmlab/mmsegmentation/blob/845098b1551608df620183da469416825b981b27/mmseg/models/segmentors/encoder_decoder.py#L155. It splits the image into grids to inference. So,in your onnx graph, you can see a large number of duplicate structures.
